src/context/reducer.js

```diff
@@ -0,0 +1,43 @@
+export default function cartReducer(state, action) {
+    switch (action.type) {
+        case "ADD_TO_CART":
+            return {
+                ...state, cartItems: [...state.cartItems, action.payload]
+            }
+        case "INCREMENT_ITEMS":
+            return {
+                ...state, cartItems: state.cartItems.map(item => item.id === action.payload.id ? {
+                    ...item,
+                    quantity: item.quantity + 1
+                } : item)
+            }
+        case "DECREMENT_ITEMS":
+            if (action.payload.quantity === 1) {
+                return { ...state, cartItems: state.cartItems.filter(cartItem => cartItem.id !== action.payload.id) }
+            } return {
+                ...state, cartItems: state.cartItems.map(item => item.id === action.payload.id ? {
+                    ...item,
+                    quantity: item.quantity - 1
+                } : item)
+            }
```

Comment:
When increasing or decreasing the quantity of items in cart, the number sometimes gets increased/decreased by 2 even on 1 click.